javascripts/discourse-kanban/initializers/initialize-discourse-kanban.js.es6

```diff
@@ -65,11 +66,30 @@ export default {
           api.modifyClass(`route:discovery.${route}`, {
             pluginId: PLUGIN_ID,
 
-            afterModel(model, transition) {
+            redirect(model, transition) {
               if (routeToBoard(transition, model.category.slug)) {
-                return this.replaceWith(
-                  `${model.category.get("url")}/l/latest?board=default`
-                );
+                // This redirect breaks the `new-topic` system, so we have to re-implement here
```

Comment:
This is kind of messy to be honest. Any way we can refactor the `new-topic` stuff out into a method that can be called or something?

Reply:
Yeah it's not ideal, but I don't think we can do much more without making core changes. Do you think it's worth refactoring core to make it easier for this theme component?

Reply:
It depends on whether any other themes/components will need to do this, or if this is truly unique?

Reply:
So far this is the only theme component I know of that hacks with the topic list in this way

Reply:
I'll leave it up to you, then.